src/main/kotlin/io/github/nomisrev/routes/tags.kt

```diff
@@ -0,0 +1,24 @@
+@file:Suppress("MatchingDeclarationName")
+
+package io.github.nomisrev.routes
+
+import io.github.nomisrev.service.TagService
+import io.ktor.server.application.Application
+import io.ktor.server.application.call
+import io.ktor.server.response.respond
+import io.ktor.server.routing.get
+import io.ktor.server.routing.route
+import io.ktor.server.routing.routing
+import kotlinx.serialization.Serializable
+
+@Serializable data class TagResponse(val tags: List<String>)
+
+fun Application.tagRoutes(tagService: TagService) = routing {
+  route("/tags") {
+    /* Registration: GET /api/tags */
+    get {
+      val tags = tagService.selectTags()
+      call.respond(TagResponse(tags))
+    }
```

Comment:
Should it be wrapped in an Either?

Reply:
This should in this case that would not be necessary, since we typically only care about wrapping _expected errors_ in `Either`.

So typically in any application there are two kind of errors:
 - expected errors, such as `UserNotFound`, or `UserAlreadyExists` and those are part of our domain logic so we want to track them with the type. These can be recovered from, or result in specific error codes, etc.
 - Unexpected errors, are those we cannot recover from. Network down, database timeout, etc and here we typically return 500.
 
 In this case `getTags` results in `emptyList` if no tags were found (valid response), or the list of actual tags. All other errors are _unexpected_ or _unrecoverable_ errors so we do not have to wrap it in `Either`.